app/src/org/commcare/android/adapters/EntitySubnodeDetailAdapter.java

```diff
@@ -32,10 +33,28 @@ public EntitySubnodeDetailAdapter(Context context, Detail detail, List<TreeRefer
         this.context = context;
         this.detail = detail;
         this.references = references;

```

Comment:
I wouldn't perform this field set if you are then going to wipe it in `initializeData`